File: 백준/Silver/2559. 수열/수열.py
# 수열
def main():
    N, K = map(int, input().split())
    temperature = list(map(int, input().split()))
    max_sum = sum(temperature[:K])
    now_sum = sum(temperature[:K])
    for i in range(1, N - K + 1):
        now_sum -= temperature[i-1]
        now_sum += temperature[i+K-1]
        max_sum = max(now_sum, max_sum)
    print(max_sum)


if __name__ == '__main__':
    main()

File: 백준/Silver/2559. 수열/수열.go
// 질문게시판에 있는 코드 오류 찾기 위해 복붙
package main
import "fmt"
import "strings"
import "bufio"
import "strconv"
import "os"

func main() {
  br := bufio.NewReader(os.Stdin)
  in, _ := br.ReadString('\n')
  in = strings.TrimSpace(in)
  nums := strings.Split(in, " ")
  N, _ := strconv.Atoi(nums[0])
  K, _ := strconv.Atoi(nums[1])
  in, _ = br.ReadString('\n')
  nums = strings.Fields(in)
  sum := make([]int, N+1)
  max := 0
  for i := 1; i <= N; i++ {
    n, _ := strconv.Atoi(nums[i-1])
    if i == 1 {
      sum[i] = n
    } else {
      sum[i] = sum[i-1] + n
    }
    if i == K {
      max = sum[i]
    } else if i > K {
      sumK := sum[i] - sum[i-K]
      if sumK > max { max = sumK }
    }
  }
  fmt.Println(max)
}